### ISSUE TITLE
Data source URL field description should remain static when Resource name is invalid

### ISSUE THREAD

**Issue by sachajudd** [silverstripe/silverstripe-ckan-registry#33]
If the 'Resource name' field is invalid the 'Data source URL' field description seems to get pushed down the page. I'd expect the field description to remain static unless the 'Data source URL' field is invalid.

![data_skitch](https://user-images.githubusercontent.com/24258161/49761701-ad23e800-fd2c-11e8-95fd-59360f28fd75.png)

**Comment by NightJar** [comment]
Just looking at this, I see a few things to note:

Moving the line above the field makes little sense to me, and is not a pattern seen anywhere else in the CMS.
However, the line is a lead in on what to do with the field - to this end it makes sense... up until data is actually entered.

So we have a a few ways to address this:
* Show the description only when the field has no data entered - in this case it will seem to have been replaced by the error messages.
* Move the field above the fields, but below the title - odd in that it doesn't match normal CMS layout
* Move the description to attach to the URL text field, as opposed to the composite of both - only downside will be that text will never flow full width on smaller screens, instead limiting itself to the width of the first field.

**Comment by robbieaverill** [comment]
I vote for the last suggestion, personally. @sachajudd may disagree, but I don't think we should be inventing new UI patterns here. Shifting the field description up so it's inline with the one on the right of it seems like the sensible choice to me

**Comment by robbieaverill** [comment]
Right, I just read this again and looked at the screenshots and I think I understand what you mean now @NightJar 😆 

@sachajudd to suggest what you'd like us to do when the field validation error shows up - maybe just hide the description and show the error instead?

![image](https://user-images.githubusercontent.com/5170590/50760662-aecbc780-1268-11e9-8936-7d1596665db6.png)

![image](https://user-images.githubusercontent.com/5170590/50760685-ba1ef300-1268-11e9-970f-df5886d86cfd.png)

**Comment by brynwhyman** [comment]
We're putting this one back into the Ready column. While it's not blocked, it requires more effort than originally expected, so other issues are taking priority.

**Comment by brynwhyman** [comment]
WIP PR at #111

**Comment by robbieaverill** [comment]
Re-triaged as medium effort, since it'll potentially require the core fieldHolder HOC to be modified or worked around
